framework/assets/yii.gridView.js

```diff
@@ -100,10 +108,12 @@
 
             $grid.find('form.gridview-filter-form').remove();
             var $form = $('<form action="' + url + '" method="get" class="gridview-filter-form" style="display:none" data-pjax></form>').appendTo($grid);
-            $.each(data, function (name, value) {
-                $form.append($('<input type="hidden" name="t" value="" />').attr('name', name).val(value));
+            $.each(data, function (name, values) {
+                $.each(values, function (index, value) {
+                    $form.append($('<input type="hidden" name="t" value="" />').attr('name', name).val(value));
```

Comment:
Did you test it? If you submit `<input name="test">` more than once, you will get the last `input` value.